app/models/service/containers.rb

```diff
@@ -24,6 +24,10 @@ def self.start_container!(wizard_state)
       end
     end
 
+    def self.pull_image(container)
+      container.compute_resource.create_image(:fromImage => container.repository_pull_url)
```

Comment:
Nods, as I mentioned it will work with a standalone registry; The general case though is talking to the hub (index), which will return the url of the registry containing the image in question (see the docs in my previous comment).